File: video-service/src/client/java/com/boclips/videos/service/client/CreateCollectionRequest.java
package com.boclips.videos.service.client;

import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

import java.util.List;

@Builder
@Data
public class CreateCollectionRequest {
    private final @NonNull String title;
    private final String description;
    private final List<String> videos;
    private final boolean isPublic;
}

File: video-service-api/src/main/kotlin/com/boclips/videos/api/request/collection/CreateCollectionRequest.kt
package com.boclips.videos.api.request.collection

import com.fasterxml.jackson.annotation.JsonSetter
import com.fasterxml.jackson.annotation.Nulls
import javax.validation.constraints.NotBlank

// TODO: why cannot we set age range and promoted?
data class CreateCollectionRequest(
    @field:NotBlank(message = "Title is required")
    val title: String?,
    val description: String? = null,
    @JsonSetter(contentNulls = Nulls.FAIL)
    val videos: List<String> = emptyList(),
    val discoverable: Boolean? = null,
    @JsonSetter(contentNulls = Nulls.FAIL)
    val subjects: Set<String> = emptySet()
)
